src/OptionT.ts

```diff
@@ -83,22 +83,69 @@ export default abstract class OptionT<T> {
    * containing `message` if this [[OptionT]] is a `None` or a default `message` if one is
    * not provided.
    *
+   * #### Note ####
+   * This function will throw an error if you attempt to call it before first checking
+   * if the [[OptionT]] is a `Some` value.
+   *
    * ```
    * const maybeOne = OptionT.some(1);
-   * // this won't throw, because it's a Some value.
+   * // this will throw, because we haven't yet checked if `maybeOne` is a `Some` value
    * const one = maybeOne.unwrap('could not unwrap a Some');
    *
+   * if (maybeOne.isSome()) {
+   *   // this will not throw, because we just confirmed it's a `Some`
+   *   const oneAgain = maybeOne.unwrap('could not unwrap a Some');
+   * }
+   *
    * // but:
    * const maybeTwo = OptionT.none();
-   * // this will throw, because it's a None value.
+   * // this will still throw, because we haven't checked what it is.
    * const two = maybeTwo.unwrap('could not unwrap a Some');
+   *
+   * if (maybeTwo.isSome()) {
+   *   // safe to unwrap, won't throw an error; also won't run because maybeTwo is a `None`
+   *   const twoAgain = maybeTwo.unwrap('could not unwrap a Some');
+   * } else {
+   *   // this will throw, because `None` values cannot be unwrapped
+   *   const twoAgain = maybeTwo.unwrap('could not unwrap a Some');
+   * }
    * ```
    *
    * @param {string} message
    * @returns {T}
    */
   abstract unwrap(message?: string): T;
 
+  /**
+   * Returns the value contained by this [[OptionT]] if it is a `Some`.  Throws an error
+   * containing `message` if this [[OptionT]] is a `None` or a default `message` if one is
+   * not provided.
+   *
+   * #### Note ####
+   * This function is not recommended for normal/production use.  It is almost always
+   * preferable to use [[OptionT.unwrap]] after first having confirmed that your [[OptionT]]
+   * is a `Some` value.
+   *
+   * This function is provided for the sole purpose of quick prototyping.  As such, this function
```

Comment:
Maybe lets say this is a utility function and not prescribe specifically what it should be used for.

Reply:
agree here, the language I used is probably too strong